docs/deprecated: legacy links env vars (#6263)
docker/cli: docs/deprecated.md

| Removed    | [`--run` flag on `docker commit`](#--run-flag-on-docker-commit)                                                                    | v0.10      | v1.13  |
| Removed    | [Three arguments form in `docker import`](#three-arguments-form-in-docker-import)                                                  | v0.6.7     | v1.12  |

### Legacy links environment variables

**Deprecated in Release: v29.0**
**Target For Removal In Release: v30.0**

Containers attached to the default bridge network can specify "legacy links" (e.g.
using `--links` on the CLI) to get access to other containers attached to that
network. The linking container (i.e. the container created with `--links`) automatically
gets environment variables that specify the IP address and port mappings of the linked
container. However, these environment variables are prefixed with the linked
container's names, making them impractical.

> Starting in Docker v29.0, these environment variables are deprecated, and won't be
> added auomtatically anymore. Users who still depend on them can start Docker Engine

Reviewer: ```suggestion
Starting with Docker v29.0, these environment variables are deprecated and will no longer be added automatically. Users who still depend on them can start Docker Engine
```
